overlord/auth/auth.go

```diff
@@ -117,6 +125,56 @@ func User(st *state.State, id int) (*UserState, error) {
 	return nil, fmt.Errorf("invalid user")
 }
 
+// Device returns the device details from the state.
+func Device(st *state.State) (*DeviceState, error) {
+	var authStateData AuthState
+
+	err := st.Get("auth", &authStateData)
+	if err == state.ErrNoState {
+		return &DeviceState{}, nil
```

Comment:
This was deliberate; unlike functions like User, there is always conceptually a DeviceState, even if it has nothing configured. I don't think it's useful to force callsites to instantiate a DeviceState manually just because they're the first place that's tried to set something.